config/features/config.go

```diff
@@ -56,7 +56,7 @@ type Flags struct {
 	AttestTimely bool // AttestTimely fixes #8185. It is gated behind a flag to ensure beacon node's fix can safely roll out first. We'll invert this in v1.1.0.
 
 	EnableSlasher                           bool // Enable slasher in the beacon node runtime.
-	EnableSlashingProtectionPruning         bool // EnableSlashingProtectionPruning for the validator client.
+	EnableSlashingProtectionPruning         bool // Enable slashing protection pruning for the validator client.
```

Comment:
If someone runs with this flag, will the pruning happen before the check for the schema? that is, will this be a viable low-risk way of switching schemas for a user?

Reply:
As written in the PR description:

> For safety purpose, if the flag `--enable-minimal-slashing-protection-database` is set while a complete slashing protection is detected, then the following warning log is displayed:

```
Minimal slashing protection database requested, while complete slashing protection database currently used. Will continue to use complete slashing protection database.
```

> and the database will be considered as complete.

> This safety net is important, since running the validator client with the flag --enable-minimal-slashing-protection-database while a complete database is already present will eventually wipe all the history specific to complete database to keep only minimal informations.

==> So no pruning will happen at all if the users asks for a minimal DB and the VC already has a complete DB.

Also:

> A tool allowing a user to convert a complete database to a minimal one will be written in a future PR. Something like:

```
prysm.sh validator slashing-protection-history convert-to-minimal
```

(I'm currently writing this one.)

Reply:
I didn't mean the new flag but `EnableSlashingProtectionPruning`

Reply:
Ohh I see sorry, I read too fast.
The pruning is done before the check of the schema.

IMO these are 2 orthogonal actions.

| minimal requested | minimal actual | pruning requested | result     |
|-------------------|----------------|--------------------|---------|
|  false                      | false                   | false                        | Nothing (Complete DB used) |
|  false                      | false                   | true                        | Pruning  (Complete DB used) |
|  false                      | true                   | false                        | Nothing (Complete DB used) |
|  false                      | true                   | true                        | Pruning, but nothing to prune actually, (Complete DB used) |
|  true                      | false                   | false                        | Will consider DB as complete |
|  true                      | false                   | true                        | Will consider DB as complete and will prune it |
|  true                      | true                   | false                        | Will consider DB as minimal and will prune it, but nothing to prune actually |